account_cutoff_prepaid/models/account_cutoff.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError, ValidationError
+
+
+class AccountCutoff(models.Model):
+    _inherit = 'account.cutoff'
+
+    @api.model
+    def _get_default_source_journals(self):
+        res = []
+        cutoff_type = self.env.context.get('cutoff_type')
+        mapping = {
+            'prepaid_expense': 'purchase',
+            'prepaid_revenue': 'sale',
+            }
+        if cutoff_type in mapping:
+            src_journals = self.env['account.journal'].search(
+                [('type', '=', mapping[cutoff_type])])
+            if src_journals:
+                res = src_journals.ids
+        return res
+
+    source_journal_ids = fields.Many2many(
+        'account.journal', column1='cutoff_id', column2='journal_id',
+        string='Source Journals', readonly=True,
+        default=lambda self: self._get_default_source_journals(),
+        states={'draft': [('readonly', False)]})
+    forecast = fields.Boolean(
+        string='Forecast',
+        readonly=True, states={'draft': [('readonly', False)]},
+        help="The Forecast mode allows the user to compute "
+        "the prepaid revenue/expense between 2 dates in the future.")
+    start_date = fields.Date(string='Start Date')
+    end_date = fields.Date(string='End Date')
```

Comment:
The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    end_date = fields.Date()
```